pkg/storage/tsdb/caching_bucket.go

```diff
@@ -111,7 +114,7 @@ func CreateCachingBucket(chunksConfig ChunksCacheConfig, metadataConfig Metadata
 	cfg := cache.NewCachingBucketConfig()
 	cachingConfigured := false
 
-	chunksCache, err := createCache("chunks-cache", chunksConfig.Backend, chunksConfig.Memcached, logger, reg)
+	chunksCache, err := createCache("chunks-cache", chunksConfig.Backend, chunksConfig.Memcached, chunksConfig.Redis, logger, reg)
```

Comment:
nit: 
Seems we can have an interface to pass only the `chunksConfig` or `metadataConfig` instead of passing all configs one by one as parameters?

I'm afraid we have another type of cache (local memory?) and we need yea another parameter

Reply:
Good idea! I have update the code to just use `CacheBackend` instead.

```
type CacheBackend struct {
	Backend   string                `yaml:"backend"`
	Memcached MemcachedClientConfig `yaml:"memcached"`
	Redis     RedisClientConfig     `yaml:"redis"`
}
```